### PR TITLE
chore(no_issue): allow to specify git ref

### PR DESCRIPTION
**Pull request by prein**
When the workflow is used withing PR context checking out default branch may not be desired. This change is meant to allow passing specific git_ref

### CODE REVIEW

File: .github/workflows/deploy.yaml [hunk 1]
```diff
@@ -41,12 +41,22 @@ on:
         required: false
         description: GHA Runner label. You may need one with custom tools on it for example.
         default: "kubernetes"
+      git_ref:
+        type: string
+        required: false
+        description: A git ref to checkout, defaults to default branch HEAD
+        default: null
```
Comment (bpleines) [inline .github/workflows/deploy.yaml:48]: Not sure how setting this to `default: null` matches the desired behavior of default = `HEAD` in the description. You mind elaborating there?

File: .github/workflows/deploy.yaml [hunk 2]
```diff
@@ -41,12 +41,22 @@ on:
...
+
 
 jobs:
   list-regions:
     runs-on: ubuntu-latest
     steps:
-      - uses: actions/checkout@v3
+      - name: Checkout
```
Comment (123BLiN) [inline .github/workflows/deploy.yaml:55]: I don't think we need to checkout here at all, but I'm not sure, maybe you can check?

File: .github/workflows/deploy.yaml (hunk 1 above)
Comment (prein) [reply .github/workflows/deploy.yaml:48]: My intention was to leave it to `[actions/checkout](https://github.com/actions/checkout/blob/5c3ccc22eb2c950a0fa5bc7c47190d8e3f7e681a/src/input-helper.ts#L60)` to render the default as they may be using "smarter" logic than just defaulting to HEAD. I'll update the description

File: .github/workflows/deploy.yaml (hunk 2 above)
Comment (prein) [reply .github/workflows/deploy.yaml:55]: you are correct, removing

**Review by 123BLiN — APPROVED** [review]

**Pull request merged**